helm-charts/splunk-otel-collector/templates/config/_common.tpl

```diff
@@ -391,6 +391,9 @@ splunk_hec/platform_metrics:
   token: "${SPLUNK_PLATFORM_HEC_TOKEN}"
   index: {{ .Values.splunkPlatform.metricsIndex | quote }}
   source: {{ .Values.splunkPlatform.source | quote }}
+  {{- if .Values.splunkPlatform.metricsSourceType }}
+  sourcetype: {{ .Values.splunkPlatform.metricsSourceType | quote }}
+  {{- end }}
```

Comment:
I feel like it may get confusing since we are adding a sourcetype to logs with "resource/logs" processor. Maybe we could create something similar for metrics/traces?

Reply:
@sbylica-splunk I can work on that.
On the other hand we set all other Splunk metadata for logs/metrics/traces in the hec exporters, so it's also quite an intuitive place for setting the sourcetype as well.
What do you think?

Reply:
Yeah, that makes more sense in my opinion. Do mind though that this would add a sourcetype everywhere - to the gateway and receiver too. If that's okay then I think we can move it to hec_exporters

Reply:
I think there are still some gaps here for telemetry data support for a `splunk-platform-enable-all-telemetry` values configuration. 
- Can you create a example in the /examples dir so we can see the for differences this PR would add? I think it would would be worth having an example where all telemetry types are enabled for `splunkPlatform.*`. This will likely make reviews easier as well.
- You can also even add unit tests under https://github.com/signalfx/splunk-otel-collector-chart/tree/main/test for unit validation tests to help ensure added Helm template logic.